rn-web/src/index.rs

```diff
@@ -0,0 +1,39 @@
+use actix_web::{Json, Query};
+use std::collections::HashMap;
+use rn_dictionary::{self, Case};
+use param;
+use super::Response;
+
+#[derive(Deserialize)]
+pub struct Params {
+    shas: param::CSV,
+    format: Option<param::Format>,
+}
+
+#[derive(Serialize)]
+pub struct BulkNames {
+    names: HashMap<String, Option<String>>,
+}
+
+impl BulkNames {
+    fn new(names: HashMap<String, Option<String>>) -> Self {
+        Self { names }
+    }
+
+    fn from_list(case: Case, shas: &[String]) -> Self {
+        let mut map = HashMap::new();
+        for sha in shas {
```

Comment:
Rather than a `for` could you use a `map` here?

Reply:
No, you can't map from a list to a hashmap. However, `fold` would be possible, just didn't seem necessary for something so simple.